src/world/Action/EffectResult.cpp

```diff
@@ -47,14 +48,36 @@ void EffectResult::damage( uint32_t amount, Common::ActionHitSeverityType severi
   m_type = Common::ActionEffectType::Damage;
 }
 
-void EffectResult::heal( uint32_t amount, Sapphire::Common::ActionHitSeverityType severity )
+void EffectResult::heal( uint32_t amount, Sapphire::Common::ActionHitSeverityType severity, bool isSelfHeal )
 {
   m_severity = severity;
   m_value = amount;
+  m_flag = isSelfHeal ? 0x80 : 0; // flag == 0x80 displays healing text at source actor
```

Comment:
probably want to typedef these as there'd be other flags that are of interest to us too

    if( isSelfHeal )
      m_flag |= ShowFlyOverOnSource

makes a bit more sense too, given how it's also used elsewhere

Reply:
I guess we don't want two healing functions so I merged them and let the caller decide who to heal and where to show flying text, same applied to mp restore and damage.